src/main/java/com/fasterxml/jackson/databind/introspect/AnnotatedConstructor.java

```diff
@@ -177,7 +178,8 @@ public String toString() {
 
     @Override
     public int hashCode() {
-        return _constructor.getName().hashCode();
+        // _constructor is nullable, so it is calculated with util
+        return Objects.hashCode(_constructor);
```

Comment:
Because of the possibility of `NullPointerException`, it has been replaced with a generic implementation.

Reply:
I don't think `null` is legal for `_constructor`: I'd rather add null check.